transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -38,18 +39,28 @@ internal class MessagingClientImpl(
     private val token: String,
     private val attachmentHandler: AttachmentHandler,
     private val messageStore: MessageStore,
+    private val reconnectionHandler: ReconnectionHandler,
+    private val stateMachine: StateMachine = StateMachineImpl(log.withTag(LogTag.STATE_MACHINE)),
 ) : MessagingClient {
     private var shouldConfigureAfterConnect = false
 
-    override var currentState: State = State.Idle
-        private set(value) {
-            if (field != value) {
-                field = value
-                stateListener?.invoke(value)
-            }
+    override val currentState: State
+        get() {
+            return stateMachine.currentState
         }
 
+    @Deprecated("Use onStateChanged() instead", ReplaceWith("onStateChanged"))
     override var stateListener: ((State) -> Unit)? = null
+        set(value) {
+            stateMachine.stateListener = value
+            field = value
+        }
+
+    override var onStateChanged: ((StateChange) -> Unit)? = null
```

Comment:
I think we should be consistent in the naming of the event listeners that MessagingClient exposes. Thus, `stateChangeListener` would match the suffix currently used with `messageListener`.